### DISCUSSION TITLE
Arduino UNO with Control_Surface library is not recognized by windows

### DISCUSSION THREAD

**Discussion by schieslu** [tttapa/Control-Surface#881] [Q&A]
Hi, i am trying to build a midi controller with an arduino UNO and some rotary encoders. For testing purposes I flashed the following sketch on the arduino: 
`#include <Control_Surface.h>
 
USBMIDI_Interface usbmidi;
const int speedMultiplier = 1;
 
CCRotaryEncoder enc {
  {5, 10},                                // pins
  MIDI_CC::General_Purpose_Controller_5, // address
  speedMultiplier,                       // multiplier
  4,                                     // pulses per click
};
 
void setup() {
  Control_Surface.begin();
  pinMode(LED_BUILTIN, OUTPUT);
}
 
void loop() { // Refresh all inputs
  Control_Surface.loop();
  digitalWrite(LED_BUILTIN, digitalRead(11));
}`

The code compiles, and I get no errors. However, the arduino is not recognized as a midi controller by windows. Am I missing something? Any help would be appreciated

**Comment by Trickster-git** [comment]
You need use hairless midi and midi loop it conect 16u2 to pc by serial, because uno don`t have native usb device  support, for convet uno to usb midi device you need use firmware for chip 16u2 and flip software to download it. 
If you want use usb device support try 32u4 chip leonardo and pro micro, or pi pico (my choice, arduio have small memory value).

I put it here
http://wiki.amperka.ru/%D1%83%D1%81%D1%82%D0%B0%D0%BD%D0%BE%D0%B2%D0%BA%D0%B0-%D0%B8-%D0%BD%D0%B0%D1%81%D1%82%D1%80%D0%BE%D0%B9%D0%BA%D0%B0-arduino-ide:arduino-uno-update-16u2?ysclid=ldhd8skk1d295593112
